apps/login/app/passkeys/verify/route.ts

```diff
@@ -0,0 +1,49 @@
+import { getSession, server, verifyPasskeyRegistration } from "#/lib/zitadel";
+import { getSessionCookieById } from "#/utils/cookies";
+import { NextRequest, NextResponse, userAgent } from "next/server";
+
+export async function POST(request: NextRequest) {
+  const body = await request.json();
+  if (body) {
+    let { passkeyId, passkeyName, publicKeyCredential, sessionId } = body;
+
+    if (!!!passkeyName) {
+      const { browser, device, os } = userAgent(request);
+      passkeyName = `${device.vendor ?? ""} ${device.model ?? ""}${
+        device.vendor || device.model ? ", " : ""
+      }${os.name}${os.name ? ", " : ""}${browser.name}`;
+    }
+    const sessionCookie = await getSessionCookieById(sessionId);
+
+    const session = await getSession(
+      server,
+      sessionCookie.id,
+      sessionCookie.token
+    );
```

Comment:
Is this an unhandled promise rejection?

Reply:
if no session could be retrieved for the context a 500 is returned below.. Any ideas on how to improve this? 🤔

Reply:
something like catching the error and returning this specific error to the user. But we anyway have another issue for this...